src/states/relocating_node.rs

```diff
@@ -39,6 +39,16 @@ use std::{
 /// Total time to wait for `RelocateResponse`.
 const RELOCATE_TIMEOUT: Duration = Duration::from_secs(60 + RESOURCE_PROOF_DURATION.as_secs());
 
+pub struct RelocatingNodeDetails {
+    pub action_sender: RoutingActionSender,
+    pub cache: Box<Cache>,
+    pub crust_service: Service,
+    pub full_id: FullId,
+    pub min_section_size: usize,
+    pub proxy_pub_id: PublicId,
+    pub timer: Timer,
+}
+
 pub struct RelocatingNode {
     action_sender: RoutingActionSender,
```

Comment:
Wouldn't it make sense to just use composition to avoid repeating all these fields?